### ISSUE TITLE
Error in Dataset finalization in a local run using OutputFileDatasetConfig. 

### ISSUE THREAD

**Issue by JoelCanteroGCO** [Azure/azure-sdk-for-python#23431]
- **Package Name**: azureml-sdk 
- **Package Version**:  1.39
- **Operating System**: Windows 10
- **Python Version**: 3.9.7

**Describe the bug**
Executing a local run with **ScriptRunConfig** object, specifying 'local' as a **compute_target** and setting some dataset arguments in **command** parameter, it throws an error: 

```
Dataset finalization failed: UserErrorException:
	Message: Request failed (400): Error when parsing request; unable to deserialize request body
	InnerException None
	ErrorResponse 
{
    "error": {
        "code": "UserError",
        "message": "Request failed (400): Error when parsing request; unable to deserialize request body"
    }
}
```

With this warning message:

```
ERROR:: DatasetFinalization  failed. . Exception Details:Traceback (most recent call last):
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/context_managers.py", line 508, in __exit__
    self.datasets.__exit__(*args)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 188, in __exit__
    self._upload(key, value)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 491, in _upload
    self._register_output(name, registered_destination, config)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 551, in _register_output
    dataset = dataset._register(self._workspace, name, description, tags, True)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/abstract_dataset.py", line 777, in _register
    raise result
azureml.exceptions._azureml_exception.UserErrorException: UserErrorException:
	Message: Request failed (400): Error when parsing request; unable to deserialize request body
	InnerException None
	ErrorResponse 
{
    "error": {
        "code": "UserError",
        "message": "Request failed (400): Error when parsing request; unable to deserialize request body"
    }

```

We set the OutputFileDatasetConfig in this way:

OutputFileDatasetConfig(name='DatosPreparados', destination=(out_ds, out_dir)).register_on_complete(name='DatosPreparados').as_upload(overwrite=True))

We have run local experiments with this setting until now and it has been perfectly working. But suddenly it does not work anymore. It is not relationed with the new version of SDK (1.39).

When we execute this experiment in an Azure compute it works perfectly and the OutputFileDataset can be registered with no errors.

I attach the complete log file.

Thank you very much,
Joel
```

[2022-03-08T14:59:39.766547] Entering context manager injector.
[2022-03-08T14:59:44.164754] context_manager_injector.py Command line Options: Namespace(inject=['ProjectPythonPath:context_managers.ProjectPythonPath', 'Dataset:context_managers.Datasets', 'RunHistory:context_managers.RunHistory', 'TrackUserError:context_managers.TrackUserError'], invocation=['python -m modeling --inference DatasetConsumptionConfig:DatosMensajes DatasetConsumptionConfig:DatosSiniestro DatasetConsumptionConfig:DatosHistoricos DatasetOutputConfig:Resultados'])
[2022-03-08T14:59:47.791] Initialize DatasetContextManager.
Script type = COMMAND
[2022-03-08T14:59:47.792620] Command=python -m modeling --inference DatasetConsumptionConfig:DatosMensajes DatasetConsumptionConfig:DatosSiniestro DatasetConsumptionConfig:DatosHistoricos DatasetOutputConfig:Resultados
[2022-03-08T14:59:47.799] Enter __enter__ of DatasetContextManager
[2022-03-08T14:59:47.801] SDK version: azureml-core==1.39.0 azureml-dataprep==2.27.1. Session id: 671ebebe-4f82-41e6-9c9a-1d5543643976. Run id: InferenceExperiment_1646751570_6a8db901.
[2022-03-08T14:59:47.801] Processing 'Resultados'.
[2022-03-08T14:59:47.801] Mode: 'upload'.
[2022-03-08T14:59:47.801] Path on compute is specified: '/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/Resultados_workspaceblobstore'.
[2022-03-08T14:59:47.904] Exit __enter__ of DatasetContextManager
[2022-03-08T14:59:47.905011] Entering Run History Context Manager.
[2022-03-08T14:59:48.503133] Command Working Directory=/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901
[2022-03-08T14:59:48.503568] Starting Linux command : python -m modeling --inference 08df8319-2003-43d6-98f0-9733454117dc 763a06f9-7b50-416f-b404-9167fcd8ccc0 0877a969-2faa-4c53-afae-61238c1b4bd6 /var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/Resultados_workspaceblobstore
    Cargando modelo...
Obteniendo datos...
    DatosHistoricos...
    DatosMensajes...
    DatosSiniestro...
Preprocesando...
    Preprocesando datos...
        DatosSiniestro
            Fechas...
            Tipos...
            Códigos...
            Territorio...
            CIF...
            Apertura...
        DatosHistoricos
        DatosMensajes...
    Combinando...
    Features derivadas...
    Validando...
Preproceso finalizado.
Iniciando step...
    Vocabulario cargado.
    Tokenizer cargado.
    Cargando modelo...
2022-03-08 16:01:49.087764: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-03-08 16:01:49.133333: I tensorflow/compiler/xla/service/service.cc:171] XLA service 0x7fbee6177820 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2022-03-08 16:01:49.133563: I tensorflow/compiler/xla/service/service.cc:179]   StreamExecutor device (0): Host, Default Version

If you want XLA:CPU, do one of the following:

 - set the TF_XLA_FLAGS to include "--tf_xla_cpu_global_jit", or
 - set cpu_global_jit to true on this session's OptimizerOptions, or
 - use experimental_jit_scope, or
 - use tf.function(jit_compile=True).

To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a
proper command-

[2022-03-08T15:02:55.641100] Command finished with return code 0


[2022-03-08T15:02:55.643283] The experiment completed successfully. Finalizing run...
[2022-03-08T15:02:55.643396] Start FinalizingInRunHistory
[2022-03-08T15:02:55.643612] Logging experiment finalizing status in history service.
Starting the daemon thread to refresh tokens in background for process with pid = 8941
Cleaning up all outstanding Run operations, waiting 300.0 seconds
2 items cleaning up...
Cleanup took 0.6705758571624756 seconds
[2022-03-08T15:03:00.106] Enter __exit__ of DatasetContextManager
[2022-03-08T15:03:00.106] Uploading output 'Resultados'.
[2022-03-08T15:03:11.340970] Logging warning in history service: ERROR:: DatasetFinalization  failed. . Exception Details:Traceback (most recent call last):
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/context_managers.py", line 508, in __exit__
    self.datasets.__exit__(*args)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 188, in __exit__
    self._upload(key, value)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 491, in _upload
    self._register_output(name, registered_destination, config)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 551, in _register_output
    dataset = dataset._register(self._workspace, name, description, tags, True)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/abstract_dataset.py", line 777, in _register
    raise result
azureml.exceptions._azureml_exception.UserErrorException: UserErrorException:
	Message: Request failed (400): Error when parsing request; unable to deserialize request body
	InnerException None
	ErrorResponse 
{
    "error": {
        "code": "UserError",
        "message": "Request failed (400): Error when parsing request; unable to deserialize request body"
    }
}

[2022-03-08T15:03:11.559583]Hosttool Job Error file path is not set
Traceback (most recent call last):
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/context_manager_injector.py", line 452, in <module>
    execute_with_context(cm_objects, options.invocation)
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/context_manager_injector.py", line 293, in execute_with_context
    print("Error occurred when setting run status to Finalizing: {}".format(ex))
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/_vendor_contextlib2.py", line 393, in __exit__
    _reraise_with_existing_context(exc_details)
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/_vendor_contextlib2.py", line 261, in _reraise_with_existing_context
    raise exc_details[1]
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/_vendor_contextlib2.py", line 382, in __exit__
    if cb(*exc_details):
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/_vendor_contextlib2.py", line 314, in _exit_wrapper
    return cm_exit(cm, *exc_details)
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/context_manager_injector.py", line 85, in __exit__
    self.context_manager.__exit__(*exc_details)
  File "/private/var/folders/wz/mrx5zx_n27x5wxmp5hf4sbsm0000gn/T/azureml_runs/InferenceExperiment_1646751570_6a8db901/azureml-setup/context_managers.py", line 508, in __exit__
    self.datasets.__exit__(*args)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 188, in __exit__
    self._upload(key, value)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 491, in _upload
    self._register_output(name, registered_destination, config)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/context_managers.py", line 551, in _register_output
    dataset = dataset._register(self._workspace, name, description, tags, True)
  File "/Users/francescsalvador/opt/miniconda3/envs/sin009/lib/python3.9/site-packages/azureml/data/abstract_dataset.py", line 777, in _register
    raise result
azureml.exceptions._azureml_exception.UserErrorException: UserErrorException:
	Message: Request failed (400): Error when parsing request; unable to deserialize request body
	InnerException None
	ErrorResponse 
{
    "error": {
        "code": "UserError",
        "message": "Request failed (400): Error when parsing request; unable to deserialize request body"
    }
}

[2022-03-08T15:03:11.798654] Finished context manager injector with Exception.
```

**Comment by catalinaperalta** [comment]
Thanks for you question @JoelCanteroGCO! Adding @bandsina who may be able to provide more insight.

**Comment by JoelCanteroGCO** [comment]
Thanks @catalinaperalta.

Any news about this? @bandsina 

Thank you,
Joel

**Comment by JoelCanteroGCO** [comment]
I have opened a Support Request on Azure because we need a solution for this issue.

**Comment by JoelCanteroGCO** [comment]
The error has been solved updating the package to azureml-core 1.40.

Thank you very much.
Joel
